one-ui/src/main/java/com/marklogic/hub/oneui/auth/LoginFailureHandler.java

```diff
@@ -40,7 +41,13 @@ public void onAuthenticationFailure(HttpServletRequest request, HttpServletRespo
         String json = new Gson().toJson(output);
         response.setContentType("application/json");
         response.setCharacterEncoding("UTF-8");
-        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        if (exception.getMessage().contains("Cannot connect to MarkLogic")) {
```

Comment:
This seems brittle - an "instanceof" check like below would be a lot better - more reliable and self-documenting. Can the custom code that creates this message throw a better exception for this class to catch?

Reply:
changed to use instanceof BadRequestException